src/utils/url.rs

```diff
@@ -0,0 +1,151 @@
+//! URL validation utilities
+//!
+//! This module provides utility functions for validating URLs.
+
+use crate::config::RpcConfig;
+use eyre::{eyre, Result};
+
+/// Validates that a URL has an HTTP or HTTPS scheme.
+///
+/// # Arguments
+/// * `url` - The URL to validate
+///
+/// # Returns
+/// * `Result<()>` - Ok if the URL has a valid scheme, error otherwise
+///
+/// # Examples
+/// ```
+/// use crate::utils::url::validate_http_url;
+///
+/// // Valid URLs
+/// assert!(validate_http_url("http://example.com").is_ok());
+/// assert!(validate_http_url("https://secure.example.com").is_ok());
+///
+/// // Invalid URLs
+/// assert!(validate_http_url("ftp://example.com").is_err());
+/// assert!(validate_http_url("invalid-url").is_err());
+/// ```
+pub fn validate_http_url(url: &str) -> Result<()> {
+    if !url.starts_with("http://") && !url.starts_with("https://") {
+        return Err(eyre!(
+            "Invalid URL scheme for {}: Only HTTP and HTTPS are supported",
+            url
+        ));
+    }
+    Ok(())
+}
+
+/// Validates all URLs in a vector of RpcConfig objects.
+///
+/// This function is a utility for validating all URLs in a configuration
+/// collection before initializing providers.
+///
+/// # Arguments
+/// * `configs` - A slice of RpcConfig objects
+///
+/// # Returns
+/// * `Result<()>` - Ok if all URLs have valid schemes, error on first invalid URL
+///
+/// # Examples
+/// ```
+/// use crate::utils::url::validate_configs_urls;
+/// use crate::config::RpcConfig;
+///
+/// let configs = vec![
+///     RpcConfig::new("https://api.example.com".to_string()),
+///     RpcConfig::new("http://localhost:8545".to_string()),
+/// ];
+/// assert!(validate_configs_urls(&configs).is_ok());
+/// ```
+pub fn validate_configs_urls(configs: &[RpcConfig]) -> Result<()> {
```

Comment:
maybe this fn can be part of RpcConfig impl.

utils url.rs is more like a generic utils file. Here we are making fn related to RpcConfig which is not generic.